components/org.wso2.carbon.identity.application.authenticator.facebook/src/main/java/org/wso2/carbon/identity/application/authenticator/facebook/FacebookAuthenticator.java

```diff
@@ -212,13 +217,21 @@ protected void processAuthenticationResponse(HttpServletRequest request,
             String code = getAuthorizationCode(request);
             String token = getToken(tokenEndPoint, clientId, clientSecret, callbackUrl, code);
 
+            claimConfig = getAuthenticatorClaimConfigurations(context);
             if (!StringUtils.isBlank(userInfoFields)) {
-                if (context.getExternalIdP().getIdentityProvider().getClaimConfig() != null && !StringUtils.isBlank
-                        (context.getExternalIdP().getIdentityProvider().getClaimConfig().getUserClaimURI())) {
-                    String userClaimUri = context.getExternalIdP().getIdentityProvider().getClaimConfig()
-                            .getUserClaimURI();
-                    if (!Arrays.asList(userInfoFields.split(",")).contains(userClaimUri)) {
+                if (!StringUtils.isBlank(claimConfig.getUserClaimURI())) {
+                    if (claimConfig == null) {
```

Comment:
You are checking for null in this line ? But above line you already use this claimConfig.getUserClaimURI()